### PR TITLE
Add two pd tests with default grace period

### PR DESCRIPTION
**Pull request by jingxu97**
Add two tests in pd.go. They are same as the flaky test, but the pod deletion has default grace period

### CODE REVIEW

File: test/e2e/pd.go [hunk 1]
```diff
@@ -134,6 +135,66 @@ var _ = framework.KubeDescribe("Pod Disks", func() {
 		return
 	})
 
+	It("Same test as above, but remove pod with default grace period 30s [Slow]", func() {
```
Comment (matchstick) [inline test/e2e/pd.go]: Can we make this name: "Should schedule a pod w/ a RW PD, gracefully remove it, then schedule it on another host [Slow]"

File: test/e2e/pd.go [hunk 2]
```diff
@@ -134,6 +135,66 @@ var _ = framework.KubeDescribe("Pod Disks", func() {
...
+		testFileContents := fmt.Sprintf("%v", mathrand.Int())
+
+		framework.ExpectNoError(f.WriteFileViaContainer(host0Pod.Name, containerName, testFile, testFileContents))
+		framework.Logf("Wrote value: %v", testFileContents)
+
+		// Verify that disk shows up for in node 1's VolumeInUse list
+		framework.ExpectNoError(waitForPDInVolumesInUse(nodeClient, diskName, host0Name, nodeStatusTimeout, true /* shouldExist */))
+
+		By("deleting host0Pod")
+		//delete pod with default grace period 30s
```
Comment (matchstick) [inline test/e2e/pd.go]: Add white space so it's "// delete pod"

File: test/e2e/pd.go [hunk 3]
```diff
@@ -134,6 +135,66 @@ var _ = framework.KubeDescribe("Pod Disks", func() {
...
+		framework.ExpectNoError(err)
+		framework.Logf("Read value: %v", v)
+
+		Expect(strings.TrimSpace(v)).To(Equal(strings.TrimSpace(testFileContents)))
+
+		// Verify that disk is removed from node 1's VolumeInUse list
+		framework.ExpectNoError(waitForPDInVolumesInUse(nodeClient, diskName, host0Name, nodeStatusTimeout, false /* shouldExist */))
+
+		By("deleting host1Pod")
+		framework.ExpectNoError(podClient.Delete(host1Pod.Name, api.NewDeleteOptions(0)), "Failed to delete host1Pod")
```
Comment (matchstick) [inline test/e2e/pd.go]: Don't forget to add grace period here too.

Comment (matchstick) [inline test/e2e/pd.go]: Not in this PR but after it is merged we should address #28183. There we can clean up these two tests in both clarity and for maintenance.

File: test/e2e/pd.go (hunk 1 above)
Comment (jingxu97) [inline test/e2e/pd.go]: fixed

File: test/e2e/pd.go (hunk 2 above)
Comment (jingxu97) [inline test/e2e/pd.go]: fixed

**Comment by matchstick** [comment]
I added this to the 1.3 milestone as it is related to 1.3 but does not need to be cherry picked for 1.3. So I am removing the milestone addition now.

File: test/e2e/pd.go [hunk 4]
```diff
@@ -75,7 +75,7 @@ var _ = framework.KubeDescribe("Pod Disks", func() {
 	})
 
 	// Flaky-- Issue #27691
-	It("[Flaky] should schedule a pod w/ a RW PD, remove it, then schedule it on another host [Slow]", func() {
+	It("[Flaky] should schedule a pod w/ a RW PD, ungracefully remove it, then schedule it on another host [Slow]", func() {
```
Comment (jeffvance) [inline test/e2e/pd.go]: I guess it's too late to comment on this pr, but I'm curious to know what you think. It seems that the defer below should be moved to after the call to podClient.Create(). If the Create() fails then the not-yet-created pod will be try to be deleted, and the not-yet-attached  disks will try to be detached. This is probably very minor, or perhaps I'm missing something?

**Pull request merged**